be/src/exprs/dictmapping_expr.cpp

```diff
@@ -14,18 +14,36 @@
 
 #include "exprs/dictmapping_expr.h"
 
+#include "column/chunk.h"
+#include "column/column_helper.h"
+#include "runtime/runtime_state.h"
+
 namespace starrocks {
 DictMappingExpr::DictMappingExpr(const TExprNode& node) : Expr(node, false) {}
 
+Status DictMappingExpr::open(RuntimeState* state, ExprContext* context, FunctionContext::FunctionStateScope scope) {
+    RETURN_IF_ERROR(Expr::open(state, context, scope));
+    if (scope != FunctionContext::FunctionStateScope::FRAGMENT_LOCAL || !_open_rewrite) {
+        return Status::OK();
+    }
+
+    return state->mutable_dict_optimize_parser()->rewrite_expr(context, this, _output_id);
```

Comment:
Is it possible to have multiple calls to open?

Reply:
the `once_flag` can be avoid it

Reply:
FRAGMENT_LOCAL also maybe open() multiple times, like in the runtime filter.

Reply:
simply, use opened flag?

Reply:
or should put this in the prepare()?

Reply:
no, look `rewrite_expr` method, it's only execute only